boot/mbed/mcuboot_main.cpp

```diff
@@ -1,79 +1,68 @@
 /*
- * Licensed to the Apache Software Foundation (ASF) under one
- * or more contributor license agreements.  See the NOTICE file
- * distributed with this work for additional information
- * regarding copyright ownership.  The ASF licenses this file
- * to you under the Apache License, Version 2.0 (the
- * "License"); you may not use this file except in compliance
- * with the License.  You may obtain a copy of the License at
+ * Copyright (c) 2020 Embedded Planet
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
  *
- *  http://www.apache.org/licenses/LICENSE-2.0
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
  *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License
  */
 
-#include "platform/mbed_application.h"
-#include "platform/mbed_toolchain.h"
-#include "platform/mbed_assert.h"
-#include "platform/mbed_interface.h"
+#if MCUBOOT_BOOTLOADER_BUILD
 
+#include <stdlib.h>
+#include "bootutil/bootutil.h"
+#include "bootutil/image.h"
+#include "hal/serial_api.h"
 #include "mbedtls/platform.h"
+#include "mbed_application.h"
 
-#include "bootutil.h"
-
-#include "flash_map_backend.h"
-
-#if MBED_CONF_MCUBOOT_BOOTLOADER_BUILD
+#define MBED_TRACE_MAX_LEVEL TRACE_LEVEL_DEBUG
+#define TRACE_GROUP "BL"
+#include "mbed-trace/mbed_trace.h"
 
-#if !DEVICE_FLASH
```

Comment:
Is this to enable the use case where mcuboot is just used to verify the main application signature at startup and _not_ for performing in-field updates?

Reply:
Which line numbers (in the new version?) are you referring to?

We just let MCUboot's `boot_go()` to do the "magic" (update/revert as needed, or just read boot info otherwise), then we just to the user aapp.